contracts/competition/Competition.sol

```diff
@@ -0,0 +1,135 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.6.0;
+
+import "../referrals/interfaces/IReferralStorage.sol";
+import "../access/Governable.sol";
+
+contract Competition is Governable
```

Comment:
hmm with this structure, we need to create a new competition contract for each competition?

seems like it might be a bit difficult to manage, e.g. would need to update subgraphs each time

can it be a single CompetitionManager contract and each competition can have an id, start date, end date instead?

Reply:
i was thinking about it as well. at the same time high chance we will miss something for the first team competition and we'll want to add more features or update some logic. so maybe it's okay?